lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -530,6 +533,88 @@ def internal_check_future_reserved_word(o, name)
     end
   end
 
+  def internal_check_file_scope(o, name, file)
+    return if file.nil?
+
+    file_scope = @file_scope_hash[file]
```

Comment:
Have a small hangup on the name `file_scope`, it is actually the *namespace* we are dealing with here, not *scope*. So perhaps `namespace` or `file_ns` instead of `file_scope`, and `file_to_namespace` for the `file_scope_hash` makes it clearer.